common/scala/src/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -46,74 +46,58 @@ import whisk.common.LoggingMarkers
  * @param dbName the name of the database to operate on
  * @param serializerEvidence confirms the document abstraction is serializable to a Document with an id
  */
-class CouchDbRestStore[Unused, DocumentAbstraction <: DocumentSerializer](
+class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](
     dbProtocol: String,
     dbHost: String,
     dbPort: Int,
     dbUsername: String,
     dbPassword: String,
     dbName: String)(implicit system: ActorSystem, jsonFormat: RootJsonFormat[DocumentAbstraction])
-    extends ArtifactStore[Unused, DocumentAbstraction]
+    extends ArtifactStore[DocumentAbstraction]
     with DefaultJsonProtocol {
 
     protected[core] implicit val executionContext = system.dispatcher
 
     private val client: CouchDbRestClient = CouchDbRestClient.make(
         dbProtocol, dbHost, dbPort, dbUsername, dbPassword, dbName)
 
-    // To make the typechecker happy.
-    private implicit def fakeDeserializer[U <: Unused, T <: DocumentAbstraction]: Deserializer[U, T] = (x: U) => ???
-
     override protected[database] def put(d: DocumentAbstraction)(implicit transid: TransactionId): Future[DocInfo] = {
-        val serialized = d.serialize().get
-        require(serialized != null, "doc undefined after serialization")
-        serialized.confirmId
-        info(this, s"[PUT] '$dbName' saving document: '${serialized.docinfo}'", LoggingMarkers.DATABASE_SAVE_START)
-
-        // The spray-json version. This seems to be missing some important fields?
-        val asJson = jsonFormat.write(d).asJsObject
-
-        // The GSON version:
-        val asGson = {
-            import spray.json._
-            import DefaultJsonProtocol._
-            import com.google.gson.Gson
-            val srzd: String = new Gson().toJson(serialized)
-            srzd.parseJson.asJsObject
-        }
+        // Uses Spray-JSON only.
+        val asJson = d.toDocumentRecord
 
-        // We'll activate this warning once we start getting rid of GSON.
-        // It doesn't really matter for now, because we insert from the GSON view
-        // anyway.
-        if (false && asJson != asGson) {
-            warn(this, s"[PUT] JSON/GSON mismatch:")
-            warn(this, s"[PUT]   - json : " + asJson)
-            warn(this, s"[PUT]   - gson : " + asGson)
-        }
+        val id: String = asJson.fields("_id").convertTo[String].trim
```

Comment:
is `_id` guaranteed to exist? could this throw an exception before the require below?